spec/PeriodicBackgroundSync-index.bs

```diff
@@ -92,3 +94,60 @@ Fetch requests within the <code>periodicsync</code> event while [=in the backgro
+
+## {{PeriodicSyncManager}} interface ## {#periodicsyncmanager-interface}
+<script type="idl">
+[Exposed=(Window,Worker)]
+interface PeriodicSyncManager {
+    Promise<void> register(DOMString tag, optional BackgroundSyncOptions options);
+    Promise<sequence<DOMString>> getTags();
+    Promise<void> unregister(DOMString tag);
+};
+
+dictionary BackgroundSyncOptions {
+    [EnforceRange] unsigned long long minInterval = 0;
+};
+</script>
+
+The <code><dfn method for=PeriodicSyncManager title="register(tag, options)">register(<var>tag</var>, <var>options</var>)</dfn></code> method, when invoked, MUST return <a>a new promise</a> <var>promise</var> and run the following steps <a>in parallel</a>:
+
+1. Let <var>serviceWorkerRegistration</var> be the {{PeriodicSyncManager}}'s associated [=service worker registration=].
+1. If |serviceWorkerRegistration|’s [=active worker=] is null, [=reject=] |promise| with an {{InvalidStateError}} and abort these steps.
+1. if the {{PermissionState}} for a {{PermissionDescriptor}} with {{name}} <code>"periodic-background-sync"</code> is not {{granted}}, [=reject=] |promise| with a {{NotAllowedError}} and abort these steps.
+1. Let <var>currentRegistration</var> be the [=periodicsync registration=] in |serviceWorkerRegistration|'s [=list of periodicSync registrations=] whose [=tag=] equals <var>tag</var> if it exists, else null.
+1. If |currentRegistration| is null:
+    1. Let |newRegistration| be a new [=periodicSync registration=].
+    1. Set |newRegistration|'s associated [=tag=] to |tag|.
+    1. Set |newRegistration|'s associated [=options=] to |options|.
+    1. Set |newRegistration|'s associated [=service worker registration=] to |serviceWorkerRegistration|.
+    1. Add |newRegistration| to |serviceWorkerRegistration|'s [=list of periodicsync registrations=].
+    1. Calculate an appropriate <var>timeToFire</var> for |newRegistration| based on the [=minInterval=] value of |options|, and any other considerations the user agent has implemented per recommendations in [[#privacy]] and Resource Usage. The time interval between now and |timeToFire| MUST be greater than or equal to the [=minInterval=] specified in |options|. Set the |newRegistration|'s [=time to fire=] to |timeToFire|.
+    1. [=Resolve=] |promise|.
+1. Else:
+    1. If |serviceWorkerRegistration|'s [=registration state=] is [=waiting=], set <var>currentRegistration</var>'s [=registration state=] to [=pending=].
+    1. If |serviceWorkerRegistration|'s [=registration state=] is [=firing=],
+        1. Set |serviceWorkerRegistration|'s [=registration state=] to [=reregisteredWhileFiring=].
+        1. If the |currentRegistration|'s [=options=] are different from |options|:
```

Comment:
Implementation note: Chrome violates this too, because we always override the options, not just when the `registration state` is set to `firing`.

Reply:
good catch, fixed it!